### PR TITLE
Library

### PR DESCRIPTION
**Pull request by bennnym**
refactoring of Library is underway for tomorrow's session.

Some of the changes include:
- Added Membership Status
- Return a book behaviour
- Add/Remove a book to the library
- Add/Remove a member from the library
- ResourceManagers, MembershipManagers and BorrowingManagers.
- Tests

### CODE REVIEW

File: Myob.Fma.BookingLibrary/BorrowedItemsManagement/BorrowingManager.cs [hunk 1]
```diff
@@ -7,29 +7,23 @@
...
 
-            _borrowedItems.Add(borrowedItem);
+            _borrowedItemsHistory.Add(borrowedItem);
         }
 
         public void ReturnItem(IResource resource, IMembership membership)
         {
             var returnedItem =
-                _borrowedItems.First(i =>
+                _borrowedItemsHistory.First(i =>
```
Comment (amrreda-myob) [inline Myob.Fma.BookingLibrary/BorrowedItemsManagement/BorrowingManager.cs]: I googled it and I found the following: 
```
`FirstOrDefault()` is just like `First()` except that, if no element match the specified 
condition than it returns default value of underlying type of generic collection. 
It does not throw `InvalidOperationException` if no element found. 
```
In this case you want it to return null if it's not exist and not throw an exception.

File: Myob.Fma.BookingLibrary/BorrowedItemsManagement/BorrowingManager.cs [hunk 2]
```diff
@@ -22,9 +22,7 @@ public void AddEntryToBorrowedItemsHistory(IResource resource, IMembership membe
 
         public void ReturnItem(IResource resource, IMembership membership)
         {
-            var returnedItem =
-                _borrowedItemsHistory.First(i =>
-                    i.Resource.Id == resource.Id && i.Member.Id == membership.Id && !i.IsReturned);
+            var returnedItem = GetOutstandingBorrowedItem(resource, membership);
```
Comment (amrreda-myob) [inline Myob.Fma.BookingLibrary/BorrowedItemsManagement/BorrowingManager.cs:25]: Null Check?

File: Myob.Fma.BookingLibraryTests/LibraryBorrowingTests.cs [hunk 3]
```diff
@@ -122,7 +125,8 @@ public void Should_Throw_An_Exception_When_Resource_Becomes_Unavailable_After_It
 
             _library.BorrowItem(resourceId, membershipId);
             var exception =
```
Comment (amrreda-myob) [inline Myob.Fma.BookingLibraryTests/LibraryBorrowingTests.cs]: Do you need that `exception` variable ?

Comment (bennnym) [reply Myob.Fma.BookingLibraryTests/LibraryBorrowingTests.cs]: removed

File: Myob.Fma.BookingLibrary/BorrowedItemsManagement/BorrowingManager.cs (hunk 2 above)
Comment (bennnym) [reply Myob.Fma.BookingLibrary/BorrowedItemsManagement/BorrowingManager.cs:25]: confirmed, not needed

**Pull request merged**